src/lib/index.js

```diff
@@ -459,13 +459,39 @@ lib.noneOrAll = function(containerIn, containerOut, attrList) {
  * @param {object} cd : calcdata trace
  * @param {string} cdAttr : calcdata key
  */
-lib.mergeArray = function(traceAttr, cd, cdAttr, isNumber) {
+lib.mergeArray = function(traceAttr, cd, cdAttr, fn) {
+    var hasFn = typeof fn === 'function';
     if(lib.isArrayOrTypedArray(traceAttr)) {
         var imax = Math.min(traceAttr.length, cd.length);
-        for(var i = 0; i < imax; i++) cd[i][cdAttr] = isNumber ? +traceAttr[i] : traceAttr[i];
+        for(var i = 0; i < imax; i++) {
+            var v = traceAttr[i];
+            cd[i][cdAttr] = hasFn ? fn(v) : v;
```

Comment:
I wonder what's faster 

```js
var hasFn = typeof fn === 'function';
for(var i = 0; i < N; i++) {
  arrayOut[i] = hasFn ? fn(arrayIn[i]) : arrayIn[i];
}
```

or 

```js
fn = typeof fn === 'function' ? fn : Lib.noop;
for(var i = 0; i < N; i++) {
  arrayOut[i] = fn(arrayIn[i]);
}
```

No need to change anything, I'm just thinking out-loud.

Reply:
Interesting question. And I like the minimalist version with `Lib.noop`!
That may depend on the browser.
Having two loops may also be fast.
```js 
var i = 0;
if (typeof fn === 'function') {
  for(; i < N; ++i) arrayOut[i] = fn(arrayIn[i]);
} else {
  for(; i < N; ++i) arrayOut[i] = arrayIn[i];
}
```